### PR TITLE
tests(windows): fix remaining "unixisms" that break some tests

### PR DESCRIPTION
**Pull request by kiike**
When I was developing an earlier PR, I wanted to test it on a vanilla installation of Windows, just to make sure no config could pollute the results.

Turns out, some tests fail on this kind of setup. Even with a benign but creepy Windows error popup during `tests/commands/open`.
image

Some reasons for the failing tests:

- symlinks no functional unless "developer mode is active" (I guess CI never sees this because it's running a modified build)
- typical unix commands such as `mv`, `echo` or `mv` not present in vanilla Windows.

Some details of this PR:

- It actually contains a fix that is present on a commit of #810, so I'll remove it from there.
- I want to explain the user why symlinks don't work if we get an exception.
- I'm sad to make `["git"] if git else []) + ["mv", folder, new_folder_path]` go, being so clever. I hope using `shutils` is OK.
- the add via symlink test is marked as xfail, just in case it succeeds.
- `script.py` now has `ls` too.

Looking forward to your feedback! I think this PR getting in will make it a bit more polished when a new user tries the 0.14 release on Windows, wants to contribute, and tests pass on a fresh install!

### CODE REVIEW

File: papis/commands/add.py
Comment (alexfikl) [inline papis/commands/add.py:431]: Does the exception not contain any useful information here? We usually do
```
logger.error(..., exc_info=exc)
```
to show the exception message or a traceback.

Comment (alexfikl) [inline papis/commands/add.py:431]: If this really needs special handling for windows, it would be better to make it a function in `papis.utils` or something so that we can reuse it in other places (`papis addto` comes to mind).

File: papis/commands/mv.py
Comment (alexfikl) [inline papis/commands/mv.py:42]: This should probably use the `mv_and_commit_resource` thing introduced in #810.

File: tests/commands/test_add.py
Comment (alexfikl) [inline tests/commands/test_add.py:193]: Just curious, but can this be checked in any way? I'm thinking we could do a `pytest.skip` instead.

File: tests/commands/test_edit.py
Comment (alexfikl) [inline tests/commands/test_edit.py:10]:
```suggestion
def get_mock_script(name: str) -> str:
```
? I'm not sure I've seed `commandlet` used anywhere. Is that a thing? :))

File: tests/commands/test_open.py
Comment (alexfikl) [inline tests/commands/test_open.py:50]: Does "start" not work here? Or?

**Review by alexfikl — APPROVED** [review]
This is great! :tada: 

Left a few questions and nitpicks, but hopefully easy to fix.

File: papis/commands/add.py
Comment (alexfikl) [reply papis/commands/add.py:431]: I can't check, but does this throw just a generic `OSError`? Maybe we can better check for a `PermissionError` or something to show that message.

**Comment by kiike** [comment]
It raises WinError, an exception that subclasses OSError. WinError can’t be in code that would run on Linux, since that’s undefined. I’ll probably move Windows specific stuff into its own utility function as you suggested so that we have specific WinError exceptions.

As for checking before actually running the operations, I guess that’s not impossible, but where should the check go? After installation, warning the user that functionality would be reduced due to dev mode not enabled, for instance?

Enric Morales ***@***.***>
________________________________

**Comment by alexfikl** [comment]
> It raises WinError, an exception that subclasses OSError. WinError can’t be in code that would run on Linux, since that’s undefined. I’ll probably move Windows specific stuff into its own utility function as you suggested so that we have specific WinError exceptions.

According to the docs, `WindowsError` is an alias of `OSError` starting with Python 3.3, not a subclass of it. But yeah, it's not defined at all on non-Windows systems, so it can't be used.

> As for checking before actually running the operations, I guess that’s not impossible, but where should the check go? After installation, warning the user that functionality would be reduced due to dev mode not enabled, for instance?

I was just thinking for that specific test, to just add a little
```
if not is_win_dev_mode():
    pytest.skip("This test requires Developer Mode on Windows")
```
But it was mostly for curiosity. They way it is now is just fine!

**Comment by kiike** [comment]
Blocked until #810 gets in.

File: papis/commands/add.py
Comment (alexfikl) [inline papis/commands/add.py:420]: Can you leave this `logger.info` message?

File: papis/commands/addto.py
Comment (alexfikl) [inline papis/commands/addto.py:101]: Leave this logger message too and make it a `logger.info` to match `papis add`.

File: papis/utils.py
Comment (alexfikl) [inline papis/utils.py:649]:
```suggestion
```

File: papis/utils.py
Comment (alexfikl) [inline papis/utils.py:652]: Maybe make this a global variable and call it `ERROR_PRIVILEGE_NOT_HELD = 1314` so that it matches the error code in the windows docs?

File: papis/utils.py
Comment (alexfikl) [inline papis/utils.py:658]: Maybe just re-raise this with additional information?
```suggestion
            raise OSError(exc.errno, 
            			 "Failed to link due to insufficient permissions. You "
                         "can try again after enabling the 'Developer mode' "
                         "and restarting.", exc.filename, exc.winerror, exc.filename2)
```
Not sure this works, but something like that..

**Review by alexfikl — COMMENTED** [review]
Left a few more suggestions, but this should be good to go once we can use the git stuff from #810.

**Comment by kiike** [comment]
I'm checking the exception on windows, to see if it is raising with all arguments

**Comment by alexfikl** [comment]
@kiike Looks good to me now! Ready to go?

**Comment by kiike** [comment]
> @kiike Looks good to me now! Ready to go?

Yes! Let's do it! But if you want me to clean up the commit list so they are logically separated by what they want to do, I can do it to ease reverting/tracking when was something introduced. Just tell me or squash merge, I don't mind :smile:

**Comment by alexfikl** [comment]
> > @kiike Looks good to me now! Ready to go?
> 
> Yes! Let's do it! But if you want me to clean up the commit list so they are logically separated by what they want to do, I can do it to ease reverting/tracking when was something introduced. Just tell me or squash merge, I don't mind 😄

I don't know, squashing should be fine. My "rule" is just that when there's some "fix this and that" commits from the review => squash it :grin:

Thank you for working on this! In it went :rocket:

**Pull request merged**
